src/ts/greedy-nav/GreedyNav.ts

```diff
@@ -0,0 +1,1062 @@
+            }
+        }
+        e = e.parentElement;
+    }
+
+    return false;
+};
+
+/**
+ * Debounced resize to throttle execution
+ * @param func
+ * @param wait
+ * @param immediate
+ * @returns {Function}
+ */
+function debounce<Return>(func: () => Return, wait: number, immediate = false) {
+    let timeout: Nullable<number>;
+    let finishedTimeout: number;
+    return function debounced(this: Return, ...args: []) {
+        // eslint-disable-next-line @typescript-eslint/no-this-alias
+        const context = this;
+
+        const later = () => {
+            timeout = null;
+            if (immediate) func.apply(context, args);
+        };
+        const callNow = immediate && timeout;
+
+        if (timeout) {
+            window.clearTimeout(timeout);
+        }
+
+        timeout = window.setTimeout(later, wait);
+        if (callNow) func.apply(context, args);
+
+        // Safari requires this to correctly fire resize when leaving fullscreen mode.
+        window.clearTimeout(finishedTimeout);
+        finishedTimeout = window.setTimeout(() => {
+            func.apply(context, args);
+        }, 800);
+    };
+}
+
+/**
+ * return true if el has a parent
+ * @param el
+ * @param parent
+ */
+const parent = (element: HTMLElement, parentNode: Node) => {
+    let el: Nullable<Node & ParentNode> = element;
+    while (el !== null) {
+        if (el.parentNode === parentNode) {
+            return true;
+        }
+        el = el.parentNode;
+    }
+    return false;
+};
+
+/**
+ * Toggle class on element
+ * @param el
+ * @param className
+ */
+const toggleClass = (el: HTMLElement, className: string) => {
+    if (el.classList) {
+        el.classList.toggle(className);
+    } else {
+        const classes = el.className.split(' ');
+        const existingIndex = classes.indexOf(className);
+
+        if (existingIndex >= 0) classes.splice(existingIndex, 1);
+        else classes.push(className);
+
+        // eslint-disable-next-line no-param-reassign
+        el.className = classes.join(' ');
+    }
+};
+
+const removeClass = (el: HTMLElement, className: string) => {
+    if (el.classList) {
+        el.classList.remove(className);
+    } else {
+        // eslint-disable-next-line no-param-reassign
+        el.className = el.className.replace(new RegExp(`s?${className}s?`), '');
+    }
+};
+
+const addClass = (el: HTMLElement, className: string) => {
+    if (el.classList) {
+        el.classList.add(className);
+    } else {
+        // eslint-disable-next-line no-param-reassign
+        el.className = `${el.className} ${className}`;
+    }
+};
+
+/**
+ * Show/hide toggle button
+ */
+export const showToggle = (
+    navWrapperElement: HTMLElement,
+    navDropdownToggleSelector: string,
+    breaks: number[]
+): void => {
+    if (breaks.length < 1) {
+        const navDropdownToggle = navWrapperElement.querySelector<HTMLElement>(
+            navDropdownToggleSelector
+        );
+
+        if (navDropdownToggle == null) {
```

Comment:
Is this `==` on purpose? Would a `if (!navnavDropdownToggle)` not work the same?

Reply:
the `==` is on purpose and `!` would work the same. I'll switch it over.